src/test/java/analyzer/AnalyzerIntegrationTest.java

```diff
@@ -149,4 +149,21 @@ void loglevels(String scenario) throws IOException {
 
         Approvals.verify(serialize(output.analysis()), Approvals.NAMES.withParameters(scenario));
     }
+
+    @ParameterizedTest
+    @ValueSource(strings = {
+            "ExemplarSolution",
+            "NotUsingBitwiseAnd",
+            "NotUsingBitwiseNot",
+            "NotUsingBitwiseOr",
+            "NotUsingBitwiseXor",
+            "NotUsingUnsignedRightShift"
```

Comment:
I'd like to see some test cases of combinations of these scenarios, or at least one that is not using any of the bitwise operators.

Reply:
Right, I see why you ask this. I left this comment before I saw the implementation for the other cases and made the assumption that they solved the exercise with something other than bitwise operators.

What I meant is a solution where more than one method is not using the expected operator. I want to cover the scenario discussed above where we don't want to leave multiple essential comments.

Hope that helps!

Reply:
I added one extra scenario that should make all essential comments trigger, but the return statements in the analyzer seem to not be working as expected and triggers all comments instead of the first one, do you have some advice to fix this? Is weird because that's how its made in the hamming and log-levels analyzers

Reply:
Then it was a good test case to add!

The reason the return statement doesn't work here is because the `visit` method is invoked separately for every method, so an early return does not prevent a second invocation.

Reply:
The simplest way would be to check if the analysis already contains an `essential` comment and if so, not add a new one.

Reply:
Thanks for the help Sander, it was pretty easy, my brain yesterday wasn't working correctly 😆